modules/widgets/milestone/milestone.php

```diff
@@ -0,0 +1,366 @@
+		add_action( 'admin_enqueue_scripts', array( __class__, 'enqueue_admin' ) );
+		add_action( 'wp_footer', array( __class__, 'localize_script' ) );
+
+		if ( is_active_widget( false, false, $this->id_base, true ) || is_active_widget( false, false, 'monster', true ) )
+			add_action( 'wp_head', array( __class__, 'styles_template' ) );
+	}
+
+	public static function enqueue_admin( $hook_suffix ) {
+		if ( 'widgets.php' == $hook_suffix )
+			wp_enqueue_style( 'milestone-admin', self::$url . 'style-admin.css', array(), '20111212' );
+	}
+
+	public static function enqueue_template() {
+		wp_enqueue_script( 'milestone', self::$url . 'milestone.js', array( 'jquery' ), '20160520', true );
+	}
+
+	public static function styles_template() {
+		global $themecolors;
+		$colors = wp_parse_args( $themecolors, array(
+			'bg'     => 'ffffff',
+			'border' => 'cccccc',
+			'text'   => '333333',
+		) );
+?>
+<style>
+.milestone-widget {
+	margin-bottom: 1em;
+}
+.milestone-content {
+	line-height: 2;
+	margin-top: 5px;
+	max-width: 17em;
+	padding: 0;
+	text-align: center;
+}
+.milestone-header {
+	background-color: <?php echo self::sanitize_color_hex( $colors['text'] ); ?>;
+	color: <?php echo self::sanitize_color_hex( $colors['bg'] ); ?>;
+	line-height: 1.3;
+	margin: 0;
+	padding: .8em;
+}
+.milestone-header .event,
+.milestone-header .date {
+	display: block;
+}
+.milestone-header .event {
+	font-size: 120%;
+}
+.milestone-countdown .difference {
+	display: block;
+	font-size: 500%;
+	font-weight: bold;
+	line-height: 1.2;
+}
+.milestone-countdown,
+.milestone-message {
+	background-color: <?php echo self::sanitize_color_hex( $colors['bg'] ); ?>;
+	border: 1px solid <?php echo self::sanitize_color_hex( $colors['border'] ); ?>;
+	border-top: 0;
+	color: <?php echo self::sanitize_color_hex( $colors['text'] ); ?>;
+	padding-bottom: 1em;
+}
+.milestone-message {
+	padding-top: 1em
+}
+</style>
+<?php
+	}
+
+	/**
+	 * Ensure that a string representing a color in hexadecimal
+	 * notation is safe for use in css and database saves.
+	 *
+	 * @param string Color in hexadecimal notation. "#" may or may not be prepended to the string.
+	 * @return string Color in hexadecimal notation on success - the string "transparent" otherwise.
+	 */
+	public static function sanitize_color_hex( $hex, $prefix = '#' ) {
+		$hex = trim( $hex );
+
+		/* Strip recognized prefixes. */
+		if ( 0 === strpos( $hex, '#' ) )
+			$hex = substr( $hex, 1 );
+		elseif ( 0 === strpos( $hex, '%23' ) )
+			$hex = substr( $hex, 3 );
+
+		if ( 0 !== preg_match( '/^[0-9a-fA-F]{6}$/', $hex ) )
+			return $prefix . $hex;
+
+		return 'transparent';
+	}
+
+	/**
+	 * Localize Front-end Script.
+	 *
+	 * Print the javascript configuration array only if the
+	 * current template has an instance of the widget that
+	 * is still counting down. In all other cases, this
+	 * function will dequeue milestone.js.
+	 *
+     * Hooks into the "wp_footer" action.
```

Comment:
It seems we switched from tabs to spaces here. Could you replace all spaces by tabs?